providers/aws/connection/connection.go

```diff
@@ -110,7 +110,7 @@ func NewAwsConnection(id uint32, asset *inventory.Asset, conf *inventory.Config)
 	c.connectionOptions = asset.Options
 	if conf.Discover != nil {
 		c.Filters = parseOptsToFilters(conf.Discover.Filter)
-		c.RegionLimits = c.Filters.GeneralDiscoveryFilters.Regions
+		c.RegionLimits = firstNonEmptyRegion(c.Filters.GeneralDiscoveryFilters.Regions, c.Filters.Ec2DiscoveryFilters.Regions)
```

Comment:
we now accommodate the case where ec2 region is set i.e
`go run apps/cnquery/cnquery.go shell aws --discover instances  --filters ec2:region=us-east-2`

Reply:
Changing this here is problematic. `c.RegionLimits` are connection wide-region limits and apply to all resources. The ec2 filter are there to just allow filtering the ec2 instances. This current change will filter globally all resources, since the `RegionLimits` is used in https://github.com/mondoohq/cnquery/blob/main/providers/aws/connection/connection.go#L304-L339.

Instead of changing it here, we want to change the code here https://github.com/mondoohq/cnquery/blob/main/providers/aws/connection/connection.go#L125-L142:

```
 		case k == "all:region":
-			d.GeneralDiscoveryFilters.Regions = append(d.GeneralDiscoveryFilters.Regions, v)
+			fallthrough
 		case k == "region":
 			d.GeneralDiscoveryFilters.Regions = append(d.GeneralDiscoveryFilters.Regions, v)
-			d.Ec2DiscoveryFilters.Regions = append(d.Ec2DiscoveryFilters.Regions, v)
```

No matter if `all:region` or `region` is used, we always set the GeneralDiscoveryFilters. Then we need to adjust the discovery logic https://github.com/mondoohq/cnquery/blob/main/providers/aws/resources/discovery.go#L114-L138 to filter by connection region limitations or ec2 region limitation

```go
	regions := []string{}
	if filters.GeneralDiscoveryFilters.Regions != nil {
		regions = append(regions, filters.GeneralDiscoveryFilters.Regions...)
	}
	if filters.Ec2DiscoveryFilters.Regions != nil {
		regions = append(regions, filters.Ec2DiscoveryFilters.Regions...)
	}

```